src/_pytest/fixtures.py

```diff
@@ -1134,20 +1092,54 @@ def fixture(
     if params is not None:
         params = list(params)
 
-    fixture_function, arguments = _parse_fixture_args(
-        callable_or_scope,
-        *args,
-        scope=scope,
-        params=params,
-        autouse=autouse,
-        ids=ids,
-        name=name,
-    )
-    scope = arguments.get("scope")
-    params = arguments.get("params")
-    autouse = arguments.get("autouse")
-    ids = arguments.get("ids")
-    name = arguments.get("name")
+    if isinstance(callable_or_scope, str):
+        fixture_function = None
+        args = (callable_or_scope, *args)
+    else:
+        fixture_function = callable_or_scope
+
+    # Positional arguments backward compatibility.
+    # If a kwarg is equal to its default, assume it was not explicitly
+    # passed, i.e. not duplicated. The more correct way is to use a
+    # **kwargs and check `in`, but that obfuscates the function signature.
+    duplicated_args = []
+    if len(args) > 0:
+        if scope == "function":
+            scope = args[0]
+        else:
+            duplicated_args.append("scope")
+    if len(args) > 1:
+        if params is None:
+            params = args[1]
+        else:
+            duplicated_args.append("params")
+    if len(args) > 2:
+        if autouse is False:
+            autouse = args[2]
+        else:
+            duplicated_args.append("autouse")
+    if len(args) > 3:
+        if ids is None:
+            ids = args[3]
+        else:
+            duplicated_args.append("ids")
+    if len(args) > 4:
+        if name is None:
+            name = args[4]
+        else:
+            duplicated_args.append("name")
+    if len(args) > 5:
+        raise TypeError(
+            "fixture() takes 5 positional arguments but {} were given".format(len(args))
+        )
+    if duplicated_args:
+        raise TypeError(
+            "The fixture arguments are defined as positional and keyword: {}. "
+            "Use only keyword arguments.".format(", ".join(duplicated_args))
+        )
+    if args:
+        warnings.warn(FIXTURE_POSITIONAL_ARGUMENTS, stacklevel=2)
+    # End backward compatiblity.
 
     if fixture_function and params is None and autouse is False:
```

Comment:
BTW, I'm interested to know why the `params is None and autouse is False` condition exists here, if anyone knows... Seems weird to just ignore `fixture_function` if `params` or `autouse` are specified?

Reply:
Judging from the `# direct decoration` comment, that's to support

```python
@pytest.fixture
def func():
    ....
```

rather than

```python
@pytest.fixture(...)
def func():
    ...
```

so there's nothing being ignored, because there are no arguments given at all in that case.

Reply:
Right, the `if fixture_function` part I understand, what I'm having trouble figuring out is the `and params is None and autouse is False` part.

The reason I care about it is that I am working on adding type-annotating `pytest.fixture()`. Without this extra `params` condition, it is straightforward to distinguish the "direct decoration" case, but with it it is more difficult.